android/src/main/java/com/frontegg/android/AuthenticationActivity.kt

```diff
@@ -103,13 +120,19 @@ class AuthenticationActivity : Activity() {
         private const val AUTH_LAUNCHED = "com.frontegg.android.AUTH_LAUNCHED"
         private const val CUSTOM_TAB_LAUNCHED = "com.frontegg.android.CUSTOM_TAB_LAUNCHED"
         private val TAG = AuthenticationActivity::class.java.simpleName
+        var onAuthFinishedCallback: (() -> Unit)? = null // Store callback
 
-        fun authenticate(activity: Activity, loginHint: String? = null) {
+        fun authenticate(
+            activity: Activity,
+            loginHint: String? = null,
+            callback: (() -> Unit)? = null
+        ) {
             val intent = Intent(activity, AuthenticationActivity::class.java)
             val authorizeUri = AuthorizeUrlGenerator().generate(loginHint = loginHint)
             intent.putExtra(AUTH_LAUNCHED, true)
             intent.putExtra(AUTHORIZE_URI, authorizeUri.first)
             intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
+            EmbeddedAuthActivity.onAuthFinishedCallback = callback
```

Comment:
Maybe here you should fill `AuthenticationActivity`.`onAuthFinishedCallback`
Not `EmbeddedAuthActivity`.`onAuthFinishedCallback`

Reply:
fixed